src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,319 @@
+            }
+
+            if (!is_writable($directory)) {
+                throw new InvalidArgumentException(sprintf('Cache directory is not writable: %s', $directory));
+            }
+        }
+
+        if ($storeSerialized) {
+            foreach ($values as $key => $value) {
+                $values[$this->validateKey($key)] = serialize($value);
+            }
+        }
+
+        $storeSerialized = var_export((bool) $storeSerialized, true);
+        $dump = str_replace("' . \"\\0\" . '", "\0", var_export($values, true));
+        unset($values);
+
+        $dump = <<<EOF
+// This file has been auto-generated by the Symfony Cache Component.
+
+\$this->values = $dump;
+
+\$this->storeSerialized = $storeSerialized;
+
+return true;
+EOF;
+
+        // Validates that the dumped code is valid (ie. no __set_state())
+        eval($dump);
+
+        $dump = "<?php\n\n".$dump."\n";
+
+        $tmpFile = tempnam(dirname($this->file), 'sf_opcache_adapter');
+
+        file_put_contents($tmpFile, $dump);
+
+        if ('\\' === DIRECTORY_SEPARATOR) {
+            // Windows does not support overwrite during rename
+            @unlink($this->file);
+            @rename($tmpFile, $this->file);
+        } else {
+            rename($tmpFile, $this->file);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        if (!$this->hasItemInternally($key)) {
+            return $this->fallbackPool->getItem($key);
+        }
+
+        if ($this->storeSerialized) {
+            $value = unserialize($this->values[$key]);
+        } else {
+            $value = $this->values[$key];
+        }
+
+        $f = $this->createCacheItem;
+
+        return $f($key, $value, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems(array $keys = array())
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        foreach ($keys as $key) {
+            $this->validateKey($key);
+        }
+
+        return $this->generateItems($keys);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem($key)
+    {
+        $this->validateKey($key);
+
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        return $this->hasItemInternally($key) || $this->fallbackPool->hasItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        $this->fallbackPool->clear();
+
+        return empty($this->values);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItem($key)
+    {
+        if ($this->hasItemInternally($key)) {
+            return false;
+        }
+
+        return $this->fallbackPool->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys)
+    {
+        $result = true;
+
+        foreach ($keys as $key) {
+            $deleted = $this->deleteItem($key);
```

Comment:
This implementation has the drawback of validating keys one by one.
Maybe we could call $this->fallbackPool->deleteItems($keys) first, then return false if it failed of if one of the keys is in `$this->values`?